governance/program/src/instruction.rs

```diff
@@ -682,27 +683,33 @@ pub fn set_governance_delegate(
     }
 }
 
-/// Creates CreateAccountGovernance instruction using optional voter weight addin
+/// Creates CreateGovernance instruction using optional voter weight addin
 #[allow(clippy::too_many_arguments)]
-pub fn create_account_governance(
+pub fn create_governance(
     program_id: &Pubkey,
     // Accounts
     realm: &Pubkey,
-    governed_account: &Pubkey,
+    governed_account: Option<&Pubkey>,
     token_owner_record: &Pubkey,
     payer: &Pubkey,
     create_authority: &Pubkey,
     voter_weight_record: Option<Pubkey>,
     // Args
     config: GovernanceConfig,
 ) -> Instruction {
-    let account_governance_address =
-        get_account_governance_address(program_id, realm, governed_account);
+    let governed_account_address = if let Some(governed_account) = governed_account {
+        *governed_account
+    } else {
+        // If the governed account is not provided then generate a unique identifier for the Governance account
+        Pubkey::new_unique()
```

Comment:
What's the use case for this? Keep in mind that `Pubkey::new_unique()` is actually deterministic, and not random, so you could easily end up with many different people creating governances on the same Pubkey.  You may actually prefer to create a random Pubkey instead, or use `Pubkey::default`

Implementation at https://github.com/solana-labs/solana/blob/17b4563a6f4f64a9d0ee321040222f9149fe12a2/sdk/program/src/pubkey.rs#L151

Reply:
It just needs `Pubkey` as the seed for the PDA if the `governed_account` is irrelevant. Is there a way to generate a unique `Pubkey` here?

Reply:
You may be better off actually creating a pubkey with random bytes, like the implementation at: https://github.com/solana-labs/solana/blob/17b4563a6f4f64a9d0ee321040222f9149fe12a2/sdk/program/src/pubkey.rs#L147 but then you can't do it on-chain, so there's that